tests/external_executor/external_executor.py

```diff
@@ -421,6 +427,105 @@ def subscribe(event_name):
     return network
 
 
+@reqs.description("Test index API")
+def test_index_api(network, args):
+    primary, _ = network.find_primary()
+
+    def add_kv_entries(network):
+        logging_executor = LoggingExecutor(primary)
+        supported_endpoints = logging_executor.supported_endpoints
+        credentials = register_new_executor(
+            primary, network, supported_endpoints=supported_endpoints
+        )
+        logging_executor.credentials = credentials
+        log_id = 14
+        with executor_thread(logging_executor):
+            with primary.client() as c:
+                for _ in range(3):
+                    r = c.post(
+                        "/app/log/public",
+                        {"id": log_id, "msg": "hello_world_" + str(log_id)},
+                    )
+                    assert r.status_code == 200
+                    log_id = log_id + 1
```

Comment:
We could avoid repeating the `14`, `3`, and `"hello_world_{N}"` formatting below if we stored them here. Build some list of "kv writes", confirm during index construction that everything in "kv writes" is received in the stream (including overwriting a previously used `log_id`!), and use that to store locally a list of "indexed entries", and finally confirm that everything in "indexed entries" (with maybe a different key from the KV key, for clarity) can be retrieved from `GetIndexedData`.